src/wazuh_db/wdb_agent.c

```diff
@@ -11,268 +11,180 @@
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    result = wdb_step(stmt) == SQLITE_DONE ? 0 : -1;
-    sqlite3_finalize(stmt);
-
-    return result;
+    return result == 0 ? 0 : -1;
 }
 
-/* Update agent version. It opens and closes the DB. Returns number of affected rows or -1 on error. */
+/* Update agent version. It opens and closes the DB. Returns 1 or -1 on error. */
 int wdb_update_agent_version(int id, const char *os_name, const char *os_version, const char *os_major, const char *os_minor, const char *os_codename, const char *os_platform, const char *os_build, const char *os_uname, const char *os_arch, const char *version, const char *config_sum, const char *merged_sum, const char *manager_host, const char *node_name, const char *agent_ip) {
     int result = 0;
-    sqlite3_stmt *stmt;
-    const char * sql = SQL_UPDATE_AGENT_VERSION;
-
-    if (wdb_open_global() < 0)
-        return -1;
-
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+  
     if(agent_ip) {
-        sql = SQL_UPDATE_AGENT_VERSION_IP;
-    }
-
-    if (wdb_prepare(wdb_global, sql, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_text(stmt, 1, os_name, -1, NULL);
-    sqlite3_bind_text(stmt, 2, os_version, -1, NULL);
-    sqlite3_bind_text(stmt, 3, os_major, -1, NULL);
-    sqlite3_bind_text(stmt, 4, os_minor, -1, NULL);
-    sqlite3_bind_text(stmt, 5, os_codename, -1, NULL);
-    sqlite3_bind_text(stmt, 6, os_platform, -1, NULL);
-    sqlite3_bind_text(stmt, 7, os_build, -1, NULL);
-    sqlite3_bind_text(stmt, 8, os_uname, -1, NULL);
-    sqlite3_bind_text(stmt, 9, os_arch, -1, NULL);
-    sqlite3_bind_text(stmt, 10, version, -1, NULL);
-    sqlite3_bind_text(stmt, 11, config_sum, -1, NULL);
-    sqlite3_bind_text(stmt, 12, merged_sum, -1, NULL);
-    sqlite3_bind_text(stmt, 13, manager_host, -1, NULL);
-    sqlite3_bind_text(stmt, 14, node_name, -1, NULL);
-    if (agent_ip) {
-        sqlite3_bind_text(stmt, 15, agent_ip, -1, NULL);
-        sqlite3_bind_int(stmt, 16, id);
+        snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_UPDATE_AGENT_VERSION_IP], os_name, os_version, os_major, os_minor,os_codename, os_platform, os_build, os_uname, os_arch, version, config_sum,merged_sum, manager_host, node_name, agent_ip , id );
     } else {
-        sqlite3_bind_int(stmt, 15, id);
+        snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_UPDATE_AGENT_VERSION],  os_name, os_version, os_major, os_minor,os_codename, os_platform, os_build, os_uname, os_arch, version, config_sum,merged_sum, manager_host, node_name , id );
     }
 
-    result = wdb_step(stmt) == SQLITE_DONE ? (int)sqlite3_changes(wdb_global) : -1;
-    sqlite3_finalize(stmt);
-
-    return result;
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
+    return result == 0 ? 1 : -1;
```

Comment:
Log information about the result and make use of `OS_SUCCESS` and `OS_INVALID`.

Reply:
Done!